### ISSUE TITLE
is the editor down or not working?

### ISSUE THREAD

**Issue by pillaroc** [fire6945/RogueEditor#17]
putting in my username and pass right but it wont let me in and it says wrong info

**Comment by realkhangninja** [comment]
have the same problem here

**Comment by RADICALDRAKE** [comment]
Yeah same here. Did anyone contact the dude who posted this on discord?

**Comment by realkhangninja** [comment]
im trying to contact him now

**Comment by RADICALDRAKE** [comment]
Great. It's strange. Pokerogue isn't down at all. I'm currently playing it and it's been fine. I thought it might be cloudflare but even on the Replit site where supposedly cloudflare shouldn't be an issue still didn't work.

**Comment by realkhangninja** [comment]
he didnt accept my friend request yet xD

**Comment by claudiunderthehood** [comment]
Hey guys! Sadly the APIs of pokerogue have been locked by a CDN so right now it's impossible to make API calls, hence, the editor for the online version is unusable. I'll try to look for a different option. For now the editor works for the offline version and you can find it here: https://github.com/claudiunderthehood/OfflineRogueEditor 

Thanks for your patience.

**Comment by fire6945** [comment]
Hey, guys.
We're very disappointed that Pokerogue's API is currently inaccessible, but I'm investigating a fix right now. Stay tuned.

**Comment by ThompsonNJ** [comment]
The API isn't inaccessible. The only thing that changed was API structure. The sessionId needs to be added on top of the auth

**Comment by realkhangninja** [comment]
can u provide us new version for it ?

> The API isn't inaccessible. The only thing that changed was the API structure. The sessionId now needs to be added on top of the auth

**Comment by ThompsonNJ** [comment]
I would think it would be easy to get all the information from the Network tab in dev tools
<img width="684" alt="image" src="https://github.com/fire6945/RogueEditor/assets/51305431/b9e559c8-0982-4576-bdec-d24f6541e3c2">
<img width="620" alt="image" src="https://github.com/fire6945/RogueEditor/assets/51305431/5293501d-055f-4bc5-b40f-42ead171521f">

`https://api.pokerogue.net/savedata/system?clientSessionId=` as an example

I don't really have the time to go through the requests more to figure out if there is a GET call for getting a user's session id though. A simple fix in the meantime while investigating further would be to prompt the user for their sessionId and require them to get it via the Network tab like I just showed.

**Comment by realkhangninja** [comment]
> clientSessionId

How long it will take mate?

**Comment by claudiunderthehood** [comment]
> > clientSessionId
> 
> How long it will take mate?

Not sure, i'll work on it this weekend whenever i can, i have uni deadlines and stuff, but we'll figure a solution :)

**Comment by realkhangninja** [comment]
> > > clientSessionId
> > 
> > How long it will take mate?
> 
> Not sure, i'll work on it this weekend whenever i can, i have uni deadlines and stuff, but we'll figure a solution :)

Damn hope u will release new patch for it soon

**Comment by ThompsonNJ** [comment]
I'm not sure if it's possible to get the sessionId without being able to connect to the db. I found the backend API endpoints that might be helpful to you but still haven't seen a sessionId endpoint. 
https://github.com/pagefaultgames/rogueserver/blob/master/api/common.go

You can look through the recent commits on the repo to see how the sessionId behavior was changed over the last 2 days.

**Comment by claudiunderthehood** [comment]
> I'm not sure if it's possible to get the sessionId without being able to connect to the db. I found the backend API endpoints that might be helpful to you but still haven't seen a sessionId endpoint. https://github.com/pagefaultgames/rogueserver/blob/master/api/common.go
> 
> You can look through the recent commits on the repo to see how the sessionId behavior was changed over the last 2 days.

Thanks for the reply! Yeah i've been scraping this for a while but no sign of the clientSessionId but no luck for now, i'll try to investigate further, when available. I'm sorry if this might take a while but that's how these things work haha.

**Comment by vasco4marques** [comment]
If I found the sessionID by loggin in in the network what do I need to update on the code? just add it in the headers after the auth as a string?

Like this??

`self.auth_headers = {
            "authorization": self.auth,
            "clientSessionId": "idStringHere",
}
`

**Comment by claudiunderthehood** [comment]
> If I found the sessionID by loggin in in the network what do I need to update on the code? just add it in the headers after the auth as a string?
> 
> 
> 
> Like this??
> 
> 
> 
> `self.auth_headers = {
> 
>             "authorization": self.auth,
> 
>             "clientSessionId": "idStringHere",
> 
> }
> 
> `

No the clientSessionId is used on the call to the APIs to GET, POST or UPDATE the data, it's not included on the headers.

**Comment by vasco4marques** [comment]
So another argument for the methods? Maybe inside de json one?
`self.clientSessionId = {
            "clientSessionId": "clientSessionIdHere"
        }`

**Comment by prephalia** [comment]
all commits related to the SessionID are

https://github.com/pagefaultgames/rogueserver/commit/2aab022ce3766f55fa4d9491eaa4f2a75da31597  - "properly use client session ID "
--
https://github.com/pagefaultgames/rogueserver/commit/c0aade2e65952fd48e0b1d44894bf2b4621d0dae - "simplify ID handling"
--
https://github.com/pagefaultgames/rogueserver/commit/436fce875901474beb2d48d7ee5676579974c37d - "add client Session ID" 

`var clientSessionId string
    if r.URL.Query().Has("clientSessionId") {
        clientSessionId = r.URL.Query().Get("clientSessionId")
    } else {
        httpError(w, r, fmt.Errorf("missing clientSessionId"), http.StatusBadRequest)
    }`

**Comment by claudiunderthehood** [comment]
> all commits related to the SessionID are
> 
> 
> 
> https://github.com/pagefaultgames/rogueserver/commit/2aab022ce3766f55fa4d9491eaa4f2a75da31597  - "properly use client session ID "
> 
> --
> 
> https://github.com/pagefaultgames/rogueserver/commit/c0aade2e65952fd48e0b1d44894bf2b4621d0dae - "simplify ID handling"
> 
> --
> 
> https://github.com/pagefaultgames/rogueserver/commit/436fce875901474beb2d48d7ee5676579974c37d - "add client Session ID" 
> 
> --
> 
> Found those
> 
> ![grafik](https://github.com/fire6945/RogueEditor/assets/170794300/fdc9ef1f-18ea-45f1-9043-a64f67344a49)
> 
> 
> 
> maybe we can simply cheese it with the legacy client ID but this will surely get removed in the future.
> 
> If you look in the commit of "simplify id handling" you see they kept a legacy-function which essentially connects without uuid and the token
> 
> 
> 
> no idea what to make of it as i have no idea of javascript so i just post my findings here
> 
> 

Thank you so much for sharing these! Yesterday I saw these and I even discovered that the clientSessionId is randomly generated by the client the moment we start the game in the browser and log in. I have a really weird workaround for this but I have to find a moment and sit down with my laptop since I'm not home this weekend.

**Comment by prephalia** [comment]
Meanwhile i found 
![grafik](https://github.com/fire6945/RogueEditor/assets/170794300/d4ccb891-8383-4c1d-a20b-391eebdf70b3)

**Comment by prephalia** [comment]
The SessionID seems to be also stored in cookies. Maybe we can work with that. phases.ts:79
![grafik](https://github.com/fire6945/RogueEditor/assets/170794300/a9b234c0-5371-40cd-8cc7-13b99ee74572)

**Comment by claudiunderthehood** [comment]
I come with good news. I was able to make a solution to get the clientSessionId and it works! Bad news is that auth_token and clientSessionId of course must coincide. Selenium for some reason in the response field when it dumps me the json of the network activity doesn't return me the token, so despite having solved a problem, I have to find a way to solve this other. One step at a time.

**Comment by prephalia** [comment]
Sounds good :) If i can help you somewhere let me know. I really wish i atleast edited the Luck of some starters before and hope it'll still work 👯

**Comment by ThompsonNJ** [comment]
> I come with good news. I was able to make a solution to get the clientSessionId and it works! Bad news is that auth_token and clientSessionId of course must coincide. Selenium for some reason in the response field when it dumps me the json of the network activity doesn't return me the token, so despite having solved a problem, I have to find a way to solve this other. One step at a time.

Oh interesting. It when I was playing around with the Network tab, my auth token and sessionId fields were always different.

**Comment by claudiunderthehood** [comment]
> > I come with good news. I was able to make a solution to get the clientSessionId and it works! Bad news is that auth_token and clientSessionId of course must coincide. Selenium for some reason in the response field when it dumps me the json of the network activity doesn't return me the token, so despite having solved a problem, I have to find a way to solve this other. One step at a time.
> 
> 
> 
> Oh interesting. It when I was playing around with the Network tab, my auth token and sessionId fields were always different.

I probably explained myself badly, when I say coincide I mean that they have to be generated in the same session, so if I use an auth token of a different session with a sessionId of another session it won't work.
